crate/operator/operations.py

```diff
@@ -557,6 +573,19 @@ async def restart_cluster(
                 patch.status["pendingPods"] = None  # Remove attribute from `.status`
                 return
         else:
+            try:
+                conn_factory = await _get_connection_factory(core, namespace, name)
+
+                await reset_cluster_setting(
```

Comment:
it means it will be reset at every retry until the cluster is healthy. Is that ok?